src/admin/admin.service.ts

```diff
@@ -0,0 +1,158 @@
+import { forwardRef, Inject, Injectable } from '@nestjs/common';
+import { v4 as uuid } from 'uuid';
+import { MailService } from '../mail/mail.service';
+import { registeredStudentInfoEmailTemplate } from '../templates/email/registered-student-info';
+import { AddRecruiterDto } from './dto/add-recruiter.dto';
+import { Recruiter } from '../recruiter/recruiter.entity';
+import { MulterDiskUploadedFiles } from '../interfaces/files';
+import * as fs from 'fs';
+import * as path from 'path';
+import { storageDir } from '../utils/storage';
+import { StudentToImport } from '../interfaces/student-to-import';
+import { isStudentToImport } from '../utils/is-student-to-import';
+import { StudentImport } from '../studentImport/studentImport.entity';
+import { DataSource } from 'typeorm';
+
+@Injectable()
+export class AdminService {
+  constructor(
+    @Inject(MailService) private mailService: MailService,
+    @Inject(forwardRef(() => DataSource)) private dataSource: DataSource,
+  ) {}
+
+  async importStudents(files: MulterDiskUploadedFiles) {
+    const fileProperty = files?.testData?.[0] ?? null;
+    const students: StudentToImport[] = [];
+
+    try {
+      if (
+        fileProperty === null ||
+        fileProperty.mimetype !== 'application/json'
+      ) {
+        throw new Error('No file or file format is different then JSON.');
+      }
+      const data = JSON.parse(
+        fs.readFileSync(
+          path.join(storageDir(), 'students-list', fileProperty.filename),
+          'utf8',
+        ),
+      );
+
+      data.forEach((element) => {
+        if (isStudentToImport(element)) {
+          students.push(element);
+        } else {
+          throw new Error('File has not proper data');
```

Comment:
Czy przypadkiem jak tu będzie błąd to nie pominie usuwanie pliku?

Reply:
Tutaj nie usunie, ale w dalej w cacth tak :)